libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3387,6 +3387,7 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 		case Type::Category::FixedBytes:
 		{
 			FixedBytesType const& from = dynamic_cast<FixedBytesType const&>(_from);
+			FixedBytesType const& to = dynamic_cast<FixedBytesType const&>(_to);
```

Comment:
You can only do this cast after the ``solAssert(toCategory == Type::Category::FixedBytes, "Invalid type conversion requested.");`` below, otherwise it will throw depending on what ``_to`` is.